src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Xunit;
+
+namespace ILVerify.Tests
+{
+    abstract class TestCase
+    {
+        public string MethodName { get; set; }
+        public ILImporter ILImporter { get; set; }
+    }
+
+    /// <summary>
+    /// Describes a test case with a method that contains valid IL
+    /// </summary>
+    class ValidILTestCase : TestCase { }
+
+    /// <summary>
+    /// Describes a test case with a method that contains invalid IL with the expected VerifierErrors
+    /// </summary>
+    class InvalidILTestCase : TestCase
+    {
+        public List<VerifierError> ExpectedVerifierError { get; set; }
+
+        /// <summary>
+        /// The expected exception during validation.
+        /// If no exception is thrown during verification this must be null.
+        /// </summary>
+        public bool ExpectsException { get; set; }
+    }
+
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class ILReader
+    {
+        public enum TestMethodType
+        {
+            Valid, Invalid
+        }
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Valid
+        /// The method must contain 1 '_'. The part before the '_' is a friendly name describing what the method does. 
+        /// The word after the '_' has to be 'Valid'. 
+        /// E.g.: 'SimpleAdd_Valid'
+        /// </summary>
+        public static IEnumerable<Object[]> GetMethodsWithValidIL()
+        {
+            var methodSelector = new Func<string[], EcmaMethod, EcmaMethodIL, TestCase>((mparams, method, methodIL) =>
+            {
+                if (mparams.Length == 2 && mparams[1].ToLower() == "valid")
+                {
+                    return new ValidILTestCase { ILImporter = new ILImporter(method, methodIL) };
```

Comment:
The ILImporter is the thing being tested. It would be more appropriate to create it during the test itself; and not in test data generator.

I think that the test data should identify the methods by `moduleName, methodHandle` or something similar.